### PR TITLE
perf(v2): improve blog mdx-loader and postcss loader

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

Resolves one TODO comment related with perf of linkify function and also removes postcss-preset-env, which we essentially don't need it, because we can use autoprefixer directly instead.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Preview

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

**Comment by lex111** [comment]
Maybe we should improve Babel somehow? For example add such a plugin? https://babeljs.io/docs/en/babel-plugin-transform-react-constant-elements

**Comment by slorber** [comment]
LGTM thanks

**Comment by slorber** [comment]
@lex111 yes there are a few babel plugins that could be useful to try.

Most of them are in https://github.com/jamiebuilds/babel-react-optimize

In the past I had problems due to the order of these plugins (https://github.com/jamiebuilds/babel-react-optimize/pull/33 ). but normally that works nicely. I think some of the plugins are even enabled in CRA now

File: packages/docusaurus/package.json [hunk 1]
```diff
@@ -93,7 +94,7 @@
     "null-loader": "^4.0.0",
     "optimize-css-assets-webpack-plugin": "^5.0.4",
     "pnp-webpack-plugin": "^1.6.4",
-    "postcss": "^8.2.6",
+    "postcss": "^8.2.7",
     "postcss-loader": "^4.1.0",
     "postcss-preset-env": "^6.7.0",
```
Comment (RDIL) [inline packages/docusaurus/package.json:99]: Does this still need to be a dependency after the update?

**Pull request merged**